### ISSUE TITLE
Surface parametrization failed

### ISSUE THREAD

**Issue by jtorresfabra** [pmp-library/pmp-library#17]
**Describe the bug**
I'm experiencing a lot of surface parametrization fails (lscm mode), even with simple meshes. At the beginning I thought it was due to non-manifold meshes. But seems is also happening with manifold ones.

**To Reproduce**

Added a patch of a bigger mesh where the parametrization is already failing.
Just load the mesh and
```
pmp::SurfaceParameterization param(mesh);
param.lscm();
```
The error I get is:
```
SurfaceParameterization: Could not solve linear system
```

Here you have the mesh:

```# OBJ export from SurfaceMesh
v -0.7163615227 0.0070284996 -0.6694365144
v -0.7187205553 0.0072185006 -0.6963074803
v -0.7367604971 0.0071144998 -0.6924955249
v -0.7344155312 0.0081265010 -0.6661835313
v -0.7174115181 0.0070054997 -0.7214004993
v -0.7372324467 0.0062004998 -0.7174305320
v -0.6941875219 0.0069044996 -0.6741214991
v -0.6956804991 0.0083875004 -0.7014294863
v -0.6478462815 0.0043691583 -0.5512301326
v -0.6589392424 0.0047934959 -0.5512301326
v -0.6554839611 0.0046613216 -0.5512301326
v -0.6709845066 0.0052074995 -0.6796914935
v -0.6720294952 0.0064525008 -0.7061825395
v -0.6515884399 0.0051004998 -0.5662325025
v -0.6721204519 0.0068165008 -0.7349734902
v -0.6477094889 0.0047705006 -0.6854125261
v -0.6487824917 0.0062985010 -0.7109814882
v -0.6387109756 0.0040080207 -0.5512301326
v -0.6491094828 0.0071565006 -0.7308275104
v -0.6297174692 0.0044315010 -0.5712625384
v -0.6245731115 0.0042155883 -0.5512301326
v -0.6189453006 0.0038505008 -0.5512301326
v -0.5997148752 0.0024406761 -0.5512301326
v -0.7305405140 0.0059715007 -0.6389564872
v -0.6013924479 0.0025071856 -0.5512301326
v -0.7119884491 0.0064955000 -0.6418855190
v -0.5865994692 0.0021995008 -0.5834735036
v -0.5927397013 0.0022911043 -0.5512301326
v -0.5778506398 0.0021319501 -0.5512301326
v -0.6900445223 0.0075905006 -0.6463505030
v -0.5618433952 0.0024415101 -0.5618459582
v -0.5618433952 0.0024919698 -0.5659798384
v -0.5618433952 0.0025049262 -0.5779944062
v -0.5618433952 0.0026342745 -0.5921446085
v -0.6675474644 0.0060415007 -0.6515235305
v -0.5709935427 0.0031215008 -0.6189585328
v -0.5618433952 0.0033831855 -0.6222074628
v -0.5618433952 0.0039914711 -0.6375563145
v -0.5618433952 0.0041960650 -0.6431018114
v -0.5755004883 0.0040725004 -0.6471894979
v -0.5618433952 0.0032420789 -0.6122533083
v -0.6449904442 0.0041095000 -0.6570414901
v -0.7356004715 0.0062144995 -0.7349734902
v -0.7491245270 0.0033355001 -0.7349734902
v -0.7491245270 0.0035375003 -0.7152515054
v -0.7491245270 0.0041765012 -0.6373705268
v -0.7478195429 0.0077884998 -0.6642045379
v -0.7491245270 0.0069535002 -0.6737325191
v -0.5924414396 0.0034595001 -0.6117604971
v -0.5971264839 0.0051975008 -0.6403545141
v -0.6946295500 0.0076505002 -0.7349734902
v -0.6139675379 0.0033884998 -0.6055175066
v -0.6188185215 0.0040125009 -0.6341854930
v -0.7491245270 0.0000000000 -0.6649144888
v -0.7491245270 0.0000000000 -0.6737325191
v -0.7491245270 0.0000000000 -0.6640315056
v -0.6355965137 0.0048645008 -0.5998325348
v -0.7491245270 0.0000000000 -0.6903535128
v -0.7491245270 0.0000000000 -0.6752654314
v -0.7491245270 0.0000000000 -0.6922904253
v -0.7491245270 0.0000000000 -0.7152515054
v -0.6406924725 0.0054564998 -0.6284515262
v -0.7491245270 0.0000000000 -0.6431018114
v -0.7491245270 0.0000000000 -0.7162207961
v -0.7491245270 0.0000000000 -0.7349734902
v -0.7339679599 0.0000000000 -0.7349734902
v -0.7159885168 0.0000000000 -0.7349734902
v -0.7356004715 0.0000000000 -0.7349734902
v -0.6946295500 0.0000000000 -0.7349734902
v -0.7140588164 0.0000000000 -0.7349734902
v -0.6923664212 0.0000000000 -0.7349734902
v -0.6721204519 0.0000000000 -0.7349734902
v -0.6700392365 0.0000000000 -0.7349734902
v -0.6491425037 0.0000000000 -0.7349734902
v -0.6554839611 0.0000000000 -0.7349734902
v -0.7484960556 0.0000000000 -0.7349734902
v -0.7491245270 0.0000000002 -0.7349734902
v -0.7264585495 0.0055155009 -0.6120445132
v -0.7455455065 0.0061045010 -0.6373324990
v -0.7431335449 0.0020305011 -0.6109035015
v -0.6627974510 0.0066045001 -0.6231265068
v -0.7062654495 0.0055065006 -0.6145415306
v -0.5726109743 0.0022295041 -0.5512301326
v -0.5618433952 0.0025308651 -0.5512301326
v -0.6846865416 0.0054835007 -0.6184875369
v -0.5618433952 0.0025762089 -0.5886166692
v -0.6033835411 0.0062074997 -0.7349734902
v -0.5792384148 0.0000000000 -0.7349734902
v -0.5808744431 0.0000000000 -0.7349734902
v -0.5618433952 0.0038664537 -0.7349734902
v -0.5618433952 0.0000000000 -0.7349734902
v -0.5618433952 0.0047717923 -0.7349734902
v -0.6015117764 0.0000000000 -0.7349734902
v -0.6033835411 0.0000000000 -0.7349734902
v -0.7175284624 0.0034545008 -0.5578095317
v -0.7223434448 0.0054145008 -0.5849955082
v -0.7419434786 0.0034414995 -0.5837435126
v -0.7355521321 0.0032537105 -0.5512301326
v -0.6959334612 0.0041405000 -0.5594105124
v -0.7011774778 0.0047705006 -0.5870954990
v -0.6737794876 0.0056325011 -0.5622094870
v -0.6793215275 0.0054465011 -0.5903185010
v -0.6072021127 0.0000000000 -0.7349734902
v -0.6075545549 0.0000000000 -0.7349734902
v -0.6244798303 0.0000000000 -0.7349734902
v -0.6261655092 0.0000000000 -0.7349734902
v -0.6470394731 0.0000000000 -0.7349734902
v -0.5618433952 0.0045137140 -0.6517116427
v -0.5785734653 0.0048855003 -0.6749745011
v -0.5618433952 0.0047299732 -0.6799957156
v -0.5618433952 0.0048280000 -0.6997354031
v -0.6712123156 0.0051663811 -0.5512301326
v -0.6795884967 0.0050035608 -0.5512301326
v -0.6941789985 0.0040637068 -0.5512301326
v -0.7003914118 0.0039176587 -0.5512301326
v -0.7161245346 0.0034299418 -0.5512301326
v -0.7205038667 0.0033654438 -0.5512301326
v -0.5618433952 0.0050669205 -0.7049937844
v -0.5801324844 0.0061655007 -0.7007204890
v -0.7491245270 0.0008645002 -0.5563964844
v -0.7454285622 0.0012955334 -0.5512301326
v -0.7491245270 0.0010908043 -0.5512301326
v -0.7491245270 0.0000000000 -0.6104965210
v -0.5618433952 0.0048167096 -0.6612778306
v -0.7491245270 0.0000000000 -0.6105730534
v -0.7491245270 0.0000000000 -0.6373705268
v -0.6005444527 0.0061544999 -0.6686275005
v -0.6024965048 0.0068635009 -0.6951605082
v -0.7491245270 0.0000000000 -0.6389039159
v -0.7491245270 0.0000000000 -0.5833604932
v -0.7491245270 0.0000000000 -0.5567328334
v -0.7491245270 0.0000000000 -0.5563964844
v -0.7491245270 0.0000000000 -0.5836412311
v -0.6225805283 0.0053805001 -0.6626535058
v -0.7491245270 0.0005596421 -0.5512301326
v -0.7491245270 0.0000000000 -0.5512301326
v -0.6249915361 0.0060364995 -0.6902045012
v -0.6259135008 0.0060035009 -0.7153034806
v -0.6032265425 0.0068354998 -0.7192485332
v -0.5618433952 0.0048708180 -0.7273030281
v -0.5618433952 0.0049936040 -0.7089808583
v -0.5618433952 0.0049470384 -0.7252701521
f 1//1 2//2 3//3
f 3//3 4//4 1//1
f 5//5 6//6 3//3
f 3//3 2//2 5//5
f 7//7 8//8 2//2
f 2//2 1//1 7//7
f 8//8 5//5 2//2
f 9//9 10//10 11//11
f 12//12 13//13 8//8
f 8//8 7//7 12//12
f 14//14 10//10 9//9
f 8//8 13//13 15//15
f 13//13 16//16 17//17
f 13//13 12//12 16//16
f 14//14 9//9 18//18
f 15//15 13//13 19//19
f 20//20 14//14 18//18
f 13//13 17//17 19//19
f 20//20 18//18 21//21
f 20//20 21//21 22//22
f 23//23 20//20 22//22
f 1//1 4//4 24//24
f 23//23 22//22 25//25
f 1//1 24//24 26//26
f 27//27 23//23 28//28
f 7//7 1//1 26//26
f 29//29 27//27 28//28
f 30//30 7//7 26//26
f 29//29 31//31 32//32
f 27//27 29//29 32//32
f 27//27 32//32 33//33
f 34//34 27//27 33//33
f 12//12 7//7 30//30
f 30//30 35//35 12//12
f 36//36 37//37 38//38
f 36//36 38//38 39//39
f 36//36 39//39 40//40
f 12//12 35//35 16//16
f 36//36 34//34 41//41
f 35//35 42//42 16//16
f 6//6 43//43 44//44
f 44//44 45//45 6//6
f 46//46 47//47 48//48
f 48//48 47//47 4//4
f 48//48 4//4 3//3
f 36//36 41//41 37//37
f 3//3 6//6 45//45
f 45//45 48//48 3//3
f 49//49 36//36 40//40
f 50//50 49//49 40//40
f 49//49 27//27 34//34
f 43//43 6//6 5//5
f 5//5 51//51 43//43
f 51//51 5//5 8//8
f 34//34 36//36 49//49
f 51//51 8//8 15//15
f 52//52 49//49 50//50
f 50//50 53//53 52//52
f 52//52 23//23 27//27
f 27//27 49//49 52//52
f 46//46 48//48 54//54
f 48//48 55//55 54//54
f 46//46 54//54 56//56
f 57//57 52//52 53//53
f 48//48 58//58 59//59
f 48//48 59//59 55//55
f 48//48 45//45 60//60
f 45//45 61//61 60//60
f 48//48 60//60 58//58
f 53//53 62//62 57//57
f 57//57 20//20 23//23
f 56//56 63//63 46//46
f 45//45 44//44 64//64
f 44//44 65//65 64//64
f 45//45 64//64 61//61
f 43//43 51//51 66//66
f 51//51 67//67 66//66
f 43//43 66//66 68//68
f 23//23 52//52 57//57
f 51//51 69//69 70//70
f 51//51 70//70 67//67
f 51//51 15//15 71//71
f 15//15 72//72 71//71
f 51//51 71//71 69//69
f 62//62 14//14 57//57
f 73//73 15//15 74//74
f 73//73 74//74 75//75
f 15//15 73//73 72//72
f 44//44 43//43 76//76
f 43//43 68//68 76//76
f 44//44 76//76 77//77
f 78//78 24//24 79//79
f 79//79 80//80 78//78
f 79//79 4//4 47//47
f 79//79 24//24 4//4
f 62//62 81//81 14//14
f 82//82 26//26 24//24
f 24//24 78//78 82//82
f 20//20 57//57 14//14
f 31//31 83//83 84//84
f 29//29 83//83 31//31
f 85//85 30//30 26//26
f 34//34 33//33 86//86
f 26//26 82//82 85//85
f 62//62 35//35 81//81
f 87//87 88//88 89//89
f 81//81 35//35 30//30
f 85//85 81//81 30//30
f 88//88 90//90 91//91
f 88//88 92//92 90//90
f 87//87 92//92 88//88
f 87//87 93//93 94//94
f 87//87 89//89 93//93
f 95//95 96//96 97//97
f 97//97 98//98 95//95
f 78//78 80//80 97//97
f 97//97 96//96 78//78
f 99//99 100//100 96//96
f 96//96 95//95 99//99
f 82//82 78//78 96//96
f 96//96 100//100 82//82
f 101//101 102//102 100//100
f 100//100 99//99 101//101
f 85//85 82//82 100//100
f 100//100 102//102 85//85
f 87//87 103//103 104//104
f 87//87 94//94 103//103
f 102//102 101//101 14//14
f 87//87 105//105 106//106
f 81//81 85//85 102//102
f 102//102 14//14 81//81
f 87//87 104//104 105//105
f 15//15 107//107 74//74
f 87//87 106//106 107//107
f 15//15 87//87 107//107
f 15//15 19//19 87//87
f 108//108 40//40 39//39
f 101//101 10//10 14//14
f 109//109 110//110 111//111
f 101//101 112//112 10//10
f 101//101 113//113 112//112
f 101//101 99//99 113//113
f 99//99 114//114 113//113
f 99//99 115//115 114//114
f 99//99 95//95 115//115
f 95//95 116//116 115//115
f 95//95 117//117 116//116
f 95//95 98//98 117//117
f 109//109 111//111 118//118
f 119//119 109//109 118//118
f 120//120 121//121 98//98
f 120//120 122//122 121//121
f 80//80 79//79 46//46
f 46//46 123//123 80//80
f 40//40 108//108 124//124
f 109//109 40//40 124//124
f 79//79 47//47 46//46
f 120//120 98//98 97//97
f 109//109 124//124 110//110
f 97//97 80//80 123//123
f 123//123 120//120 97//97
f 123//123 46//46 125//125
f 46//46 126//126 125//125
f 127//127 109//109 119//119
f 119//119 128//128 127//127
f 127//127 50//50 40//40
f 129//129 46//46 63//63
f 46//46 129//129 126//126
f 40//40 109//109 127//127
f 120//120 130//130 131//131
f 120//120 131//131 132//132
f 120//120 123//123 133//133
f 134//134 127//127 128//128
f 120//120 133//133 130//130
f 120//120 135//135 122//122
f 120//120 132//132 135//135
f 132//132 136//136 135//135
f 128//128 137//137 134//134
f 134//134 53//53 50//50
f 127//127 134//134 50//50
f 42//42 134//134 137//137
f 137//137 16//16 42//42
f 42//42 62//62 53//53
f 137//137 138//138 17//17
f 17//17 16//16 137//137
f 87//87 19//19 17//17
f 17//17 138//138 87//87
f 128//128 139//139 138//138
f 138//138 137//137 128//128
f 138//138 139//139 87//87
f 134//134 42//42 53//53
f 42//42 35//35 62//62
f 119//119 140//140 139//139
f 139//139 128//128 119//119
f 87//87 140//140 92//92
f 87//87 139//139 140//140
f 140//140 141//141 142//142
f 140//140 119//119 141//141
f 119//119 118//118 141//141
```

**Expected behavior**
Create a parametrization for the mesh.

**Environment (please complete the following information):**
 - OS: Ubuntu 16.04
 - Compiler: clang-6
 - Build type: Debug/Release

**Comment by jtorresfabra** [comment]
Additional note: Same mesh with blender unwrap (which as far as I know is using lscm too) works.

**Comment by dsieger** [comment]
Looks like @mbotsch fixed as a by-product of commit ace532d926fdc7a068e6a85027db6e879b3a46bb

Trying your example mesh with the latest version seems to give a reasonable result, see attached.
![Screenshot from 2019-09-29 17-13-07](https://user-images.githubusercontent.com/6951585/65834578-98750880-e2dc-11e9-9f6d-31bafdee7a4e.png)
